tests/rules/test_float_values.py

```diff
@@ -0,0 +1,128 @@
+# Copyright (C) 2022 the yamllint contributors
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+from tests.common import RuleTestCase
+
+
+class FloatValuesTestCase(RuleTestCase):
+    rule_id = "float-values"
+
+    def test_disabled(self):
+        conf = (
+            "float-values: disable\n"
+            "new-line-at-end-of-file: disable\n"
+            "document-start: disable\n"
+        )
+        self.check("angle: 0.0", conf)
+        self.check("angle: .NaN", conf)
+        self.check("angle: .INF", conf)
+        self.check("angle: .1", conf)
+        self.check("angle: 10e-6", conf)
+
+    def test_numeral_before_decimal(self):
+        conf = (
+            "float-values:\n"
+            "  require-numeral-before-decimal: true\n"
+            "  forbid-scientific-notation: false\n"
+            "  forbid-nan: false\n"
+            "  forbid-inf: false\n"
+            "new-line-at-end-of-file: disable\n"
+            "document-start: disable\n"
+        )
+        self.check("angle: .1", conf, problem=(1, 10))
+        self.check("angle: 0.0", conf)
+        self.check("angle: \"0.0\"", conf)
+        self.check('angle: !custom_tag 0.0', conf)
+
+    def test_scientific_notation(self):
+        conf = (
+            "float-values:\n"
+            "  require-numeral-before-decimal: false\n"
+            "  forbid-scientific-notation: true\n"
+            "  forbid-nan: false\n"
+            "  forbid-inf: false\n"
+            "new-line-at-end-of-file: disable\n"
+            "document-start: disable\n"
+        )
+        self.check("angle: 10e6", conf, problem=(1, 12))
+        self.check("angle: 10e-6", conf, problem=(1, 13))
```

Comment:
Why is the error at position 12, then at position 13?

These positions don't exist on these lines: their lengths are 11 and 12. So a visual tool that reports problems in an editor wouldn't be able to show the problems.

It would make sense to have it at position 8 (or 10, why not) in both cases. What do you think? Is it possible to fix this?

Reply:
> Why is the error at position 12, then at position 13?

I followed the implementation of `octal_values` when implementing `float_values`. The current behavior is to set `LintProblem.column` equal to the last column where the problem occurs (exclusive):

https://github.com/adrienverge/yamllint/blob/ac19d1e427747f91f287d5b6cca4eeac3a232493/yamllint/rules/octal_values.py#L115

`test_octal_values` likewise has the same behavior (column is always 1  + string length):

https://github.com/adrienverge/yamllint/blob/ac19d1e427747f91f287d5b6cca4eeac3a232493/tests/rules/test_octal_values.py#L36

Reply:
This reveals a bug in `octal-value`, thanks for noticing it!

In any case, this new rule should report the problem where it really is. Said differently, the error should be located here:
```yaml
angle: 10e-6
       ↑
     error
```

not here:
```yaml
angle: 10e-6
            ↑
          error
```

What do you think?

Reply:
Done